### PR TITLE
fork https://github.com/abitrolly/podman

### PR DESCRIPTION
**Pull request by haircommander**
per conversation in https://github.com/containers/libpod/issues/1915

Signed-off-by: Peter Hunt <pehunt@redhat.com>

### CODE REVIEW

**Comment by TomSweeneyRedHat** [comment]
@haircommander git validation is not your friend here

**Comment by haircommander** [comment]
:crossed_fingers:

File: contrib/snapcraft/README.md [hunk 1]
```diff
@@ -0,0 +1,82 @@
+<h1 align="center">
+  <img src="https://raw.githubusercontent.com/containers/podman.io/master/images/podman.svg?sanitize=true" alt="podman">
+  <br />
+</h1>
+
+<p align="center"><b>This is the snap for <a href="https://podman.io/">podman</a></b>, <i>“Managing Pods, Containers, and Container Images in Multiple Formats (Docker, OCI, ...).”</i>. It works on Ubuntu, Fedora, Debian, and other major Linux
```
Comment (TomSweeneyRedHat) [inline contrib/snapcraft/README.md]: Can we mention RHEL and/or SUSE here?

Comment (haircommander) [reply contrib/snapcraft/README.md]: fixed

**Comment by TomSweeneyRedHat** [comment]
LGTM
but I'm far from snap-literate.  Would like someone more familar with snap to take a looksee.

**Comment by haircommander** [comment]
@TomSweeneyRedHat as far as I can tell, this is a WIP. @abitrolly is that correct?

File: contrib/snapcraft/README.md (hunk 1 above)
Comment (rhatdan) [reply contrib/snapcraft/README.md]: Yes Suse, RHEL, Centos ...

**Comment by rhatdan** [comment]
@lsm5 @jnovy PTAL

**Comment by abitrolly** [comment]
Yes, it is a WIP. It might even require some hacks on `snapd` side.

**Comment by mikeroyal** [comment]
Hi @abitrolly, most of the bugs you're experiencing were fixed in snapcraft 3.o or later. When building the snap make to follow these steps for prep.

# Prep
sudo snap install snapcraft --classic
sudo snap install multipass --beta --classic

**Comment by abitrolly** [comment]
@mikeroyal I run `snapcraft` from inside container - https://github.com/snapcore/snapcraft/pull/2659

**Comment by rhatdan** [comment]
We need @mikeroyal @abitrolly @lsm5 @jnovy To approve this PR. since I have no knowlege of how Snap works.  Can we merge this into libpod and then automatically get snaps to build and be installable on Ubuntu and Debian distros?

File: contrib/snapcraft/snap/snapcraft.yaml [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+base: core18
+
+parts:
+  podman:
+    plugin: go
+    source: https://github.com/containers/libpod/archive/v0.11.1.1.tar.gz
+    go-importpath: github.com/containers/libpod
+    build-packages:
+     # https://github.com/containers/libpod/blob/master/install.md#build-and-run-dependencies
```
Comment (jnovy) [inline contrib/snapcraft/snap/snapcraft.yaml]: Used this snapcraft.yaml to build it:
```
# snapcraft --debug
...

The 'podman' part is missing libraries that are not included in the snap or base. They can be satisfied by adding the following entry for this part
stage-packages:
- libarchive13
- libassuan0
- libgpgme11
- libicu60
- libostree-1-1
- libsoup2.4-1
- libxml2
'grade' property not specified: defaulting to 'stable'
Snapping 'podman' /                                                                                                                                           
Snapped podman_0.11.1.1_amd64.snap
# ls -l
total 28640
-rw-r--r--. 1 root root 29323264 Aug 12 15:28 podman_0.11.1.1_amd64.snap
-rw-r--r--. 1 root root     1134 Aug 12 15:24 snapcraft.yaml
```

Should we add the stage-packages entry there as well?

Comment (abitrolly) [reply contrib/snapcraft/snap/snapcraft.yaml]: @jnovy most likely `podman` snap won't work without them if you install and try the snap.

Comment (mikeroyal) [reply contrib/snapcraft/snap/snapcraft.yaml]: Hi @jnovy, 

The 'stage-packages:' should be placed right below the 'build-packages:'

Comment (jnovy) [reply contrib/snapcraft/snap/snapcraft.yaml]: Added the missing stage-packages to snapcraft.yaml.

Comment (haircommander) [reply contrib/snapcraft/snap/snapcraft.yaml]: thanks for adding that @jnovy , I squashed the commits to fix the validate problem :)

Comment (jnovy) [reply contrib/snapcraft/snap/snapcraft.yaml]: lgtm now - @mheon - Could you merge this please?

**Comment by haircommander** [comment]
PTAL @mikeroyal @abitrolly

**Comment by mheon** [comment]
/approve
The 0.11.1 thing gives me pause, but we can fix that in future releases... LGTM

**Comment by haircommander** [comment]
@rhatdan @baude @TomSweeneyRedHat @vrothberg @QiWang19 PTAL

**Comment by TomSweeneyRedHat** [comment]
LGTM, but would like a mheon head nod.

**Pull request merged**